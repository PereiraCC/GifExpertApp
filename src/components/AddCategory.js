import React, { useState } from 'react'

export const AddCategory = () => {

    const [inputValue, setinputValue] = useState('Hello world');

    const handleInputChange = ( e ) => {
        setinputValue( e.target.value );
    }

    const handleSubmit = ( e ) => {
        e.preventDefault();
    
        console.log('Submit done');
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange }
                />
        </form>
    )
}
