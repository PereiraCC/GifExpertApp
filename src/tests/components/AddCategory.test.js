import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";


describe('<AddCategory /> test', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories} />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories} />);
    });
   
    test('Must show correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Must change textBox', () => {
       
        const input = wrapper.find('input');
        const value = 'Hello world';

        input.simulate('change', { target: { value } });
        expect( wrapper.find('p').text().trim() ).toBe( value );

    });

    test('should not post the information with submit', () => {
      
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();

    });
    
    test('must called setCategories and clean textbox', () => {
       
        const value = 'Hello world';

        wrapper.find('input').simulate('change', { target: { value }});
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        expect( wrapper.find('input').text().trim() ).toBe('');

    });
    
});
