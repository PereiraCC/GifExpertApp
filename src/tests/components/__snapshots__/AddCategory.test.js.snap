// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> test Must show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
